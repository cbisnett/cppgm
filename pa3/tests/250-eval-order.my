0u
0u
5u
0u
0u
5u
0u
0u
5u
0u
0u
5u
0u
0u
5u
0u
0u
5u
eof
