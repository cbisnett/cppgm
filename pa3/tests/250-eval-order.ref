5
error
1
error
0
error
5
error
1
error
0
error
5
error
1
error
0
error
eof
