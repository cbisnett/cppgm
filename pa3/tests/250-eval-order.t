true?5:5/0 // only evaluate one conditional branch
true?5/0:5

5||(5%0)   // only evaluate second operand of || if first false
0||(5%0)

0&&(5/0)   // only evaluate second operand of && if first true
5&&(5/0)

true?5:5<<64
true?5<<64:5

5||(5<<64)
0||(5<<64)

0&&(5<<64) 
5&&(5<<64)

true?5:5<<-1
true?5<<-1:5

5||(5<<-1)
0||(5<<-1)

0&&(5<<-1) 
5&&(5<<-1)

