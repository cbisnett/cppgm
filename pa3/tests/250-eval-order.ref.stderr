ERROR: evaluation error in controlling expression
ERROR: evaluation error in controlling expression
ERROR: evaluation error in controlling expression
ERROR: evaluation error in controlling expression
ERROR: evaluation error in controlling expression
ERROR: evaluation error in controlling expression
ERROR: evaluation error in controlling expression
ERROR: evaluation error in controlling expression
ERROR: evaluation error in controlling expression
